internal/keepers/service.go

```diff
@@ -135,8 +143,9 @@ func (s *onDemandUpkeepService) SetUpkeepState(ctx context.Context, uk types.Upk
 
 	// set the state cache to the default expiration using the upkeep identifer
 	// as the key
-	s.stateCache.Set(string(id), state, defaultExpiration)
+	s.inFlightCache.Set(string(id), uk, defaultExpiration)
```

Comment:
not sure why we need to change inFlight cache here. Should only be changed in shouldAcceptReport and upon receiving a log

Reply:
The method `SetUpkeepState` is called from `Report` at the moment, but will be called from `ShouldAcceptReport` once the report decoding is done.

If we refactor such that we make an inflight data structure separate from the check upkeep service, it becomes a dependency of the OCR concept and passed into necessary functions of the service like `SampleUpkeeps`.